test/tests/layer.js

```diff
@@ -82,7 +89,30 @@ suite("d3.layer", function() {
 			this.layer.draw([]);
 			assert(this.insert.calledOn(this.dataBind.returnValues[0].enter.returnValues[0]));
 		});
-		
+		test("by default removes exiting nodes from the DOM", function() {
+			this.layer.draw([1]);
+			assert.equal(this.layer.selectAll("g").size(), 1);
+			this.layer.draw([]);
+			assert.equal(this.layer.selectAll("g").size(), 0);
+		});
+		test("invokes the provided `remove` method in the context of the layer's bound 'exit' selection", function() {
+			var updating, exiting;
+
+			this.layerWithRemove.draw([1, 2, 3]);
+			this.layerWithRemove.draw([]);
+
+			updating = this.dataBind.returnValues[1];
+			exiting = updating.exit.returnValues[0];
+
+			assert(this.remove.calledOn(exiting));
```

Comment:
So is the assumption that `remove` will be defined just like `insert` and `dataBind` if a user wishes to do so? I can't think of a scenario where we would need a separate method to basically call `.remove` on a selection. I tend to call remove in my `exit` transitions - my worry is that by defining `remove` as a separate function, it will not be clear that it _only_ pertains to the `exiting` selection. My initial assumption, that this function enabled one to effectively remove the drawn layer, is probably something that a lot of people will consider it to mean as well.

What do you think?